File: BOJ/[2026] 소풍/KingDonggyu/2026.js
function solution([K, N, F], ...friend) {
  const graph = Array.from(Array(N + 1), () => []);

  friend.forEach(([f1, f2]) => {
    graph[f1].push(f2);
    graph[f2].push(f1);
  });

  graph.forEach((arr) => {
    arr = arr.sort((a, b) => a - b);
  });

  const dfs = (num, selected, visited) => {
    if (selected.length === K) {
      console.log(selected.join('\n'));
      process.exit();
    }

    for (const v of graph[num]) {
      if (visited[v]) {
        continue;
      }

      if (selected.every((s) => graph[v].includes(s))) {
        visited[v] = true;
        dfs(v, [...selected, v], visited);
      }
    }
  };

  for (let i = 1; i < N + 1; i++) {
    if (graph[i].length >= K - 1) {
      const visited = Array(N + 1).fill(false);
      visited[i] = true;
      dfs(i, [i], visited);
    }
  }

  console.log(-1);
}

const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];

rl.on('line', (line) => {
  input.push(line.split(' ').map((x) => +x));
  input.length === +input[0][2] + 1 && rl.close();
}).on('close', () => {
  solution(...input);
});

File: BOJ/[2026] 소풍/JFe/2026.py
import sys
from collections import defaultdict

def dfs(num, l):
    if len(l) == K:
        for v in l: print(v)
        exit()
    for i in range(num + 1, N + 1):
        if not visited[i]:
            for v in l:
                if v not in adj[i]: break
            else:
                visited[i] = True
                dfs(i, l+[i])

K, N, F = map(int, sys.stdin.readline().split())
adj = defaultdict(list)
# 인접 리스트 생성
for i in range(F):
    a, b = map(int, sys.stdin.readline().split())
    adj[a].append(b)
    adj[b].append(a)
# 인접 리스트 기준으로 DFS 탐색
for i in range(1, N + 1):
    visited = defaultdict(bool)
    visited[i] = True
    dfs(i, [i])
print(-1)